### DISCUSSION TITLE
containerd total downloads is invalid?

### DISCUSSION THREAD

**Discussion by pacoxu** [badges/shields#9157] [Q&A]
https://img.shields.io/github/downloads/containerd/containerd/total?style=plastic

Here in https://github.com/LinuxSuRen/hd-home/blob/master/README.md many badges are correct except containerd

**Comment by chris48s** [comment]
The reason why this is failing is because the API response is so large
If I call https://api.github.com/repos/containerd/containerd/releases?per_page=500
that is over 13Mb of JSON. We set a limit of 10Mb on the size of API response we will download to render a badge.

I've not looked into it, but I wonder if it would be possible to rewrite this badge to use the v4 GraphQL API to work this number out more efficiently? (this may or may not be possible)

**Reply by chris48s** [reply]
Had a quick look into this. Using the rest API we can request up to 500 releases in a page, meaning we can count downloads from the most recent 500 releases with one API call. If we switch to GraphQL for it, we can't ask it to sum `downloadCount`s for every `releaseAsset` for us and the the maximum number of release objects we can request in one API call is 100. Bah :(

**Reply by calebcartwright** [reply]
Given how we handle pagination in GitHub's v3 API, I suspect we could get under the size limit by lowering the per page count at the expense of having to fire off more async requests for the additional pages that would result in.

I'm not sure if it makes sense to do that by default, as I think having `per_page` set to the max is the ideal approach in the overwhelming majority of cases, so I wonder if we could add some type of `?hasBigReleases` query parameter that could be provided on the badge route wherein we'd use a lower per_page value?
